feat(logging): add standard format logging alongside JSON

collinschreyer-dev opened the pull request:
- Add standard formatter to complement existing JSON format
- Keep both formats
- Format: '{timestamp} - {level} - {message}'
- Goal: debug log visibility in cloud.gov

Adding a standard, human-readable log handler alongside the existing format means you can cross-check that logs are being sent properly in both styles. This makes it easier to pinpoint any issues related to formatting, configuration, or how cloud.gov processes logs

collinschreyer-dev committed:
feat(logging): add standard format logging alongside JSON

- Add standard formatter to complement existing JSON format
- Keep both formats
- Format: '{timestamp} - {level} - {message}'
- Goal: debug log visibility in cloud.gov

BuckinghamAJ commented on src/fbo_scraper/json_log_formatter.py:
```diff
@@ -18,19 +17,18 @@ def add_fields(self, log_record, record, message_dict):
             log_record["level"] = log_record["level"].lower()
-            if log_record["level"] == "level 12":
-                log_record["level"] = "found it top"
+        if log_record["level"] == "level 12":
```
If you are pulling it out of the initial if statement be sure to use the .get functionality or you'll end up with a Key Error.

BuckinghamAJ commented on src/fbo_scraper/json_log_formatter.py:
```diff
@@ -18,19 +17,18 @@ def add_fields(self, log_record, record, message_dict):
-            ):  # greater than DEBUG but less then INFO gets marked as debug for log searching
-                log_record["level"] = "debug"
+            # did we set the level numerically?
+            matches = re.match("level ([0-9]*)", log_record["level"])
```
I think this should be pulled back out and not under the else statement.

BuckinghamAJ commented on src/fbo_scraper/json_log_formatter.py:
```diff
@@ -45,29 +43,35 @@ def process_log_record(self, log_record):
+    logger.addHandler(json_handler)
+
+    # Add standard formatter output
+    standard_handler = logging.StreamHandler(stdout)
```
Aren't we now going to have 2 log streams happening at once? Is this what you want to check to see how it works? 

To be more dynamic, I think we need to add a command line option to spin up the JSON logging instead of the standard logging.

BuckinghamAJ reviewed (commented):
Some things to note, discuss and change for pushing to dev

collinschreyer-dev committed:
Addressed feedback on JSON log formatter

- Avoided KeyError by using log_record.get('level') in CustomJsonFormatter.add_fields.
- Moved level number matching outside the else block to ensure it always executes.
- Enabled dual logging streams to stdout to have both JSON and standard logs simultaneously.
- Acknowledged the suggestion to add a command-line option to control logging format in the future.
